execution/contextTree.sh

```diff
@@ -1500,6 +1506,88 @@ function upgrade_cmdb_repo_to_v2_0_0() {
   return $return_status
 }
 
+function upgrade_cmdb_repo_to_v2_0_1() {
+  # Reorganise state files into a directory tree based on deployment unit and placement
+  #
+  # The format of the state tree will follow the pattern
+  # state/{df}/{env}/{seg}/{du}/{placement}
+  #
+  # Delete definition files because their file name contains the occurrence name not the
+  # deployment unit. They will be regenerated into the correct dir on the next build.
+  local root_dir="$1";shift
+  local dry_run="$1";shift
+
+
+  pushTempDir "${FUNCNAME[0]}_$(fileName "${root_dir}")_XXXX"
+  local tmp_dir="$(getTopTempDir)"
+  local return_status=0
+
+  readarray -t state_dirs < <(find "${root_dir}" -type d -name "state" )
+  for state_dir in "${state_dirs[@]}"; do
+
+    readarray -t deployment_frameworks < <(find "${state_dir}" -mindepth 1 -maxdepth 1 -type d )
+    for df_dir in "${deployment_frameworks[@]}"; do
+      local deployment_framework="$(fileName "${df_dir}")"
+
+      debug "${dry_run}Checking ${deployment_framework} deployment framework ..."
+      readarray -t state_files < <(find "${df_dir}" -type f )
+      for state_file in "${state_files[@]}"; do
+
+        local state_filepath="$(filePath "${state_file}")"
+        local state_filename="$(fileName "${state_file}")"
+
+        if contains "${state_filename}" "([a-z0-9]+)-(.+)-([a-z][a-z0-9]+)-([a-z]{2}-[a-z]+-[1-9])(-pseudo)?-(.+)"; then
+          local stack_level="${BASH_REMATCH[1]}"
+          local stack_deployment_unit="${BASH_REMATCH[2]}"
+          local stack_account="${BASH_REMATCH[3]}"
+          local stack_region="${BASH_REMATCH[4]}"
+
+          case "${stack_level}" in
+            defn)
+              # Definition files are copied on every template creation
+              info "${dry_run}Deleting ${state_file} ..."
+              [[ -n "${dry_run}" ]] && continue
+              git_rm "${state_file}" || { return_status=1; }
+              ;;
+
+            *)
+              # Add deployment unit based subdirectories
+              if contains "${state_filepath}" "${stack_deployment_unit}"; then
+                debug "${dry_run}Ignoring ${state_file}, already moved ..."
+              else
+                #
+                case "${stack_region}" in
+                  us-east-1)
+                    local placement="global"
+                    ;;
+                  *)
+                    local placement="default"
+                    ;;
+                esac
+                local du_dir="${state_filepath}/${stack_deployment_unit}/${placement}"
+                info "${dry_run}Moving ${state_file} to ${du_dir} ..."
+                [[ -n "${dry_run}" ]] && continue
+                if [[ ! -d "${du_dir}" ]]; then
+                  mkdir -p "${du_dir}"
+                fi
+                git_mv "${state_file}" "${du_dir}" || { return_status=1; }
```

Comment:
On my existing files this has not determined the ```${stack_deployment_unit}``` so it has copied them to the dir state/cf/integration/default rather than individual du dirs.

Example file name: ```seg-vpc-mswdev-eastus-template.json```

It does however create correct directories for them if I recreate the template.

This move has also moved the entire contents of the directory. I had a number of unrelated files in this dir which I had meant to tidy up and I can see that they've made the move to the new directory too.

Reply:
@RossMurr4y thanks for testing on Azure

As discussed, the move is part of the CMDB v2.0.0 change and is expected. 

The failure to add the du directory was because the format of regions for Azure doesn't match that of AWS. I've added logic to change the expected file format based on deployment framework. While the AWS format is fairly unique and so anchors the regex correctly, te Azure ones are a bit more problematic. Would you mind providing the list of Azure regions you are currently using so I can look for those?